praw/objector.py

```diff
@@ -61,6 +71,41 @@ def objectify(self, data):
                 or {'id', 'name', 'permissions'}.issubset(set(data.keys()))):
             parser = self.parsers[self._reddit.config.kinds['redditor']]
             return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and (
+                {'conversation', 'messages', 'user', 'modActions'}
+                .issubset(set(data.keys()))):
+            # Modmail conversation
+            parser = self.parsers['ModmailConversation']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'actionTypeId' in data.keys():
+            # Modmail mod action
+            data = self.fix_dict_keys(data, {'actionTypeId': 'action_type'})
+            parser = self.parsers['ModmailAction']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isInternal' in data.keys():
+            # Modmail message
+            camel_attributes = {'bodyMarkdown': 'body_markdown',
+                                'isInternal': 'is_internal'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            parser = self.parsers['ModmailMessage']
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'isAdmin' in data.keys():
+            # Modmail author
+            camel_attributes = {'isAdmin': 'is_admin',
+                                'isDeleted': 'is_deleted',
+                                'isHidden': 'is_hidden',
+                                'isMod': 'is_mod',
+                                'isOp': 'is_op',
+                                'isParticipant': 'is_participant'}
+            data = self.fix_dict_keys(data, camel_attributes)
+            del(data['id'])
+            parser = self.parsers[self._reddit.config.kinds['redditor']]
+            return parser.parse(data, self._reddit)
+        elif isinstance(data, dict) and 'displayName' in data.keys():
+            # Modmail subreddit
+            data = self.fix_dict_keys(data, {'displayName': 'display_name'})
+            parser = self.parsers[self._reddit.config.kinds[data['type']]]
+            return parser.parse(data, self._reddit)
         elif isinstance(data, dict) and 'user' in data:
```

Comment:
This method is turning out to be a monster. It's okay for now, but if you can think of a better way to handle it, please do.